v2/cmd/asoctl/cmd/template.go

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package cmd
+
+import (
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/Azure/azure-service-operator/v2/cmd/asoctl/internal/template"
+)
+
+type templateOptions struct {
+	raw        bool
+	crdPattern []string
+}
+
+// newTemplateCommand creates a new cobra Command when invoked from the command line
+func newTemplateCommand() (*cobra.Command, error) {
+	var options templateOptions
+
+	cmd := &cobra.Command{
+		Use:   "template <path to ASO YAML>",
+		Short: "Template creates a YAML file from the specified ASO yaml template",
+		Example: `asoctl template https://github.com/Azure/azure-service-operator/releases/download/v2.6.0/azureserviceoperator_v2.6.0.yaml --crd-pattern "resources.azure.com/*;containerservice.azure.com/*;keyvault.azure.com/*;managedidentity.azure.com/*;eventhub.azure.com/*"
```

Comment:
Would be good if user doesn't have to specify the whole url. We can just ask them for the version they want, and construct a URL in asoctl?

Reply:
Changed this